File: Daily Test/Python/Find Matrix Size 27.8.2025.py
import sys

lines = [list(map(int, line.split())) for line in sys.stdin.read().strip().splitlines()]

rows, cols = 0, len(lines[0])
for row in lines:
    if len(row) == cols:
        rows += 1
    else:
        print(rows, cols)
        rows, cols = 1, len(row)
print(rows, cols)

File: Daily Test/Java/Find Matrix Size 27.8.2025.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<String> lines = new ArrayList<>();
        while (sc.hasNextLine()) {
            String s = sc.nextLine().trim();
            if (s.isEmpty()) break;
            lines.add(s);
        }
        int rows = 0, cols = -1;
        for (String line : lines) {
            String[] parts = line.split("\\s+");
            int count = parts.length;

            if (cols == -1) {
                cols = count;
                rows = 1;
            } else if (count == cols) {
                rows++;
            } else {
                System.out.println(rows + " " + cols);
                cols = count;
                rows = 1;
            }
        }
        System.out.println(rows + " " + cols);
    }
}